server.js

```diff
@@ -21,7 +15,78 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send("Avocado sales data api. Endpoints: /api, dynamic routes: /api/region, /api/region/id. Query parameter: /api?date=");
+});
+
+app.get("/api", (req, res) => {
+  const { date } = req.query;
+
+  let allAvocadoSalesData = avocadoSalesData;
+
+  if (date) {
+    allAvocadoSalesData = allAvocadoSalesData.filter((salesData) => 
+      salesData.date === date
+    );
+  }
+
+  res.status(200).json({
+    data: allAvocadoSalesData,
+    success: true
+  });
+});
+
+app.get("/api/:regions", (req, res) => {
+  const { regions } = req.params;
+
+  //how to make this dataPerRegion accessible in the next app.get?
+  const dataPerRegion = avocadoSalesData.filter((avocadoData) => 
+    avocadoData.region.toLowerCase() === regions.toLowerCase()
+  );
+
+  //regions if statements for printing
+  if (dataPerRegion && dataPerRegion.length !== 0) {
+    res.status(200).send({
+      data: dataPerRegion,  
+      success: true
+    });
+  } else if (dataPerRegion.length === 0) {
+    res.status(200).send({
+      data: 'Response is successful but no region with that name exists',  
+      success: true
+    });
+  } else {
+    res.status(404).send({
+      data: 'Not found',
+      success: false
+    });
```

Comment:
I think this `else` code will never be used. I think the comparison can be tweaked a little as well. Either the data (array) has a length greater than 0 which also means it exists, or it is an empty array. Tried testing a bit and can't get a scenario where it runs. So I think the code can be revised to just and `if-else` using the successful but empty response, removing the "Not found" In lecture I think they said we can think of Not Found being for when we're filtering, vs when we're finding like you do later in your ID.
```
  if (dataPerRegion.length !== 0) {
    res.status(200).send({
      data: dataPerRegion,
      success: true,
    });
  } else {
    res.status(200).send({
      data: "Response is successful but no region with that name exists",
      success: true,
    });
  }